commit 6afd1198d111ccff4c91be96d8c631afc29325e7
Author: Henry Ptasinski <henryp@broadcom.com>
Date:   Sun Sep 26 13:19:44 2010 -0700

    staging: brcm80211: Make compiling of brcm80211.ko and brcmfmac.ko mutually exclusive.
    
    Temporary fix until utils get cleaned up.
    
    Signed-off-by: Henry Ptasinski <henryp@broadcom.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- a/drivers/staging/brcm80211/Kconfig
+++ b/drivers/staging/brcm80211/Kconfig
@@ -1,15 +1,33 @@
 menuconfig BRCM80211
-	tristate "Broadcom 802.11 WLAN driver for bcm43xx chips"
-	depends on PCI && WLAN
-	select WIRELESS_EXT
-	select WEXT_PRIV
-	default N
+	tristate "Broadcom IEEE802.11n WLAN drivers"
+	depends on WLAN
+
+choice
+	prompt "Broadcom IEEE802.11n driver style"
+	depends on BRCM80211
 	help
-	  If built as a module, it will be called brcm80211.ko.
+	Select the appropriate driver style from the list below.
 
 config BRCM80211_PCI
-	bool "Broadcom 802.11 WLAN NIC driver"
-	depends on BRCM80211
-	default y
+	bool "Broadcom IEEE802.11n PCIe SoftMAC WLAN driver"
+	depends on PCI
+	depends on BRCM80211 && MAC80211
+	select FW_LOADER
+	---help---
+	  This module adds support for PCIe wireless adapters based on Broadcom
+	  IEEE802.11n SoftMAC chipsets.  If you choose to build a module, it'll
+	  be called brcm80211.ko.
 
-source "drivers/staging/brcm80211/brcmfmac/Kconfig"
+config BRCMFMAC
+	bool "Broadcom IEEE802.11n embedded FullMAC WLAN driver"
+	depends on MMC
+	depends on BRCM80211 && CFG80211
+	select FW_LOADER
+	select WIRELESS_EXT
+	select WEXT_PRIV
+	---help---
+	  This module adds support for embedded wireless adapters based on
+	  Broadcom IEEE802.11n FullMAC chipsets.  This driver uses the kernel's
+	  wireless extensions subsystem.  If you choose to build a module,
+	  it'll be called brcmfmac.ko.
+endchoice
--- a/drivers/staging/brcm80211/brcmfmac/Makefile
+++ b/drivers/staging/brcm80211/brcmfmac/Makefile
@@ -56,6 +56,7 @@ DHDOFILES = dhd_linux.o linux_osl.o bcmu
 	dhd_linux_sched.o dhd_cdc.o bcmsdh_sdmmc.o bcmsdh.o bcmsdh_linux.o	\
 	bcmsdh_sdmmc_linux.o
 
-obj-$(CONFIG_BRCMFMAC) += brcmfmac.o
+#obj-$(CONFIG_BRCMFMAC) += brcmfmac.o
+obj-m += brcmfmac.o
 brcmfmac-objs += $(DHDOFILES)
 EXTRA_LDFLAGS += --strip-debug
